### PR TITLE
Refactor analytics to not use api.gradio.app

### PR DESCRIPTION
**Pull request by freddyaboulton**
## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Instead of using `_send_telemetry_in_thread`, why not use `send_telemetry`? That way we can avoid spinning up our own thread for each analytics request and instead use the single-thread approach that is baked into `huggingface_hub` (since Lite doesn't support threads, I guess we'd still need to use `_send_telemetry_in_thread` in that case)

cc @Wauplin

**Comment by freddyaboulton** [comment]
@abidlabs _send_telemetry_in_thread does not actually launch a new thread as far as I can tell, it just sends the request to HF hub. In contrast, `send_telemetry` will place data in a queue and then start a new thread to pull data off that queue and send to the hub. I think we still need to run the analytics in a separate thread so that the "get ip address" operation does not block, so I think `_send_telemetry_in_thread` is better to not have an additional background thread running.

**Comment by abidlabs** [comment]
@freddyaboulton yes you're right, the `_send_telemetry_in_thread `, but create a thread ourselves, e.g. when we do:

```py
threading.Thread(
            target=_do_normal_analytics_request,
```

it seems like it'd be better to use the documented public method if possible?

File: gradio/analytics.py [hunk 1]
```diff
@@ -29,60 +31,91 @@ async def pyodide_pyfetch(*_args, **_kwargs):
...
+        url = f"https://huggingface.co/api/telemetry/{path}"
+        headers = build_hf_headers(
+            token=False,  # no need to send a token for telemetry
+            library_name="gradio",
+            library_version=data.get("version"),
+            user_agent=data,
+        )
         await asyncio.wait_for(
-            pyodide_pyfetch(url, method="POST", headers=headers, body=body),
+            pyodide_pyfetch(url, method="HEAD", headers=headers),
```
Comment (freddyaboulton) [inline gradio/analytics.py:114]: I originally wrote this to use `_send_telemetry_in_thread` but the analytics requests were not showing up in the webworker network tab.

If I use `pyodide_pyfetch`, they show up in the network tab. Still not sure if this works 100% though.

So I think we need this lower-level approach for wasm and can't rely on higher level functions in hub python library. 


image

**Comment by abidlabs** [comment]
Nvm @freddyaboulton you are right we will need to spin our own thread to get the IP so let's stick with `_send_telemetry_in_thread`. @Wauplin is that a safe choice or is there a better option you'd suggest?

File: gradio/analytics.py (hunk 1 above)
Comment (whitphx) [reply gradio/analytics.py:114]: If `_send_telemetry_in_thread` uses threading as its name implies, it [doesn't work](https://pyodide.org/en/stable/usage/wasm-constraints.html#included-but-not-working-modules) on Lite/Pyodide.

I think using `pyodide_pyfetch` is a right approach on Lite and it should work.

Comment (freddyaboulton) [reply gradio/analytics.py]: I reverted the changes for wasm since we can't set the user-agent on chrome!

Comment (whitphx) [reply gradio/analytics.py]: Ref: https://stackoverflow.com/a/42815264/13103190

**Comment by Wauplin** [comment]
> we will need to spin our own thread to get the IP so let's stick with `_send_telemetry_in_thread`. @Wauplin is that a safe choice or is there a better option you'd suggest?

It's a reasonably safe choice yes. Don't see a better option that does not involve re-implementing the request.
Could you add a comment in `huggingface_hub`'s `_send_telemetry_in_thread` to reference to gradio-lite so that we don't forget? Little chance we change this internal method anytime soon but better safe that sorry :)

**Review by abidlabs — APPROVED** [review]
Code changes LGTM @freddyaboulton! Jw if you checked to confirm that the analytics went through to kibana?

**Comment by freddyaboulton** [comment]
Yes analytics are going through!

**Pull request merged**